rocky/components/modal/script.js

```diff
@@ -10,67 +10,60 @@ export function initDialogs(element) {
 }
 
 export function initDialog(modal) {
-  let input_elements = [];
+  let id = window.location.toString().split("#")[1];
+  let dialog_element = modal.querySelector("dialog");
+  if (!dialog_element) return;
 
-  modal.querySelector(".modal-trigger").addEventListener("click", (event) => {
-    // Get and clone input elements to be able to "reset" them on "cancel".
-    input_elements = modal.querySelectorAll("input, textarea");
+  let trigger = document.querySelector(
+    "[data-modal-id='" + dialog_element.id + "']",
+  );
+  if (!trigger) return;
 
-    // Store the initial value in a data attribute
-    input_elements.forEach((element) => {
-      element.defaultvalue = element.value;
-    });
+  // If ID is present in the URL on init, open the dialog immediately.
+  if (id) {
```

Comment:
It might be good to move the full `window.location` check here instead of assigning to `id`, since everything else in the init function should use `dialog_element.id`.

In fact, it probably makes sense to check the `id` from the location hash against the `dialog_element.id`, to avoid potentially calling `showModal` multiple times, once for each dialog that's being initialized.

Reply:
Or it could make sense to simply move this behaviour to a `DOMContentLoaded` event listener.